### PR TITLE
chore(storage-monitor): improve `free_space` calculation and cli default value

### PR DESCRIPTION
**Pull request by yjhmelody**

### CODE REVIEW

File: client/storage-monitor/src/lib.rs [hunk 1]
```diff
@@ -45,19 +45,19 @@ pub struct StorageMonitorParams {
 	#[arg(long = "db-storage-threshold", value_name = "MB", default_value_t = 1000)]
 	pub threshold: u64,
 
-	/// How often available space is polled.
+	/// How often available space is polled in seconds.
 	#[arg(long = "db-storage-polling-period", value_name = "SECONDS", default_value_t = 5, value_parser = clap::value_parser!(u32).range(1..))]
```
Comment (yjhmelody) [inline client/storage-monitor/src/lib.rs:49]: I think the default `5` is not good. `6` or `12` is better idea.

File: client/storage-monitor/src/lib.rs [hunk 2]
```diff
@@ -117,10 +117,10 @@ impl StorageMonitorService {
 
 	/// Returns free space in MB, or error if statvfs failed.
 	fn free_space(path: &Path) -> Result<u64, Error> {
-		Ok(fs4::available_space(path).map(|s| s / 1_000_000)?)
+		Ok(fs4::available_space(path).map(|s| s / 1024 / 1024)?)
```
Comment (davxy) [inline client/storage-monitor/src/lib.rs:120]: I think that the returned value was purposely returned in MB and not in MiB.
But maybe @michalkucharczyk knows better

File: client/storage-monitor/src/lib.rs (hunk 1 above)
Comment (davxy) [reply client/storage-monitor/src/lib.rs:49]: Why 6 instead of 5?

File: client/storage-monitor/src/lib.rs (hunk 2 above)
Comment (michalkucharczyk) [reply client/storage-monitor/src/lib.rs:120]: Well I made this decision basing on https://en.wikipedia.org/wiki/Megabyte being `10^6` not `1024^2`.
Feel free to change it if you really feel like we need to use `MiB` (Mebibyte). But please also adjust `clap` `args` documentation.

Comment (davxy) [reply client/storage-monitor/src/lib.rs:122]: If we return the value in MiB then also change the docs (MB -> MiB) please

File: client/storage-monitor/src/lib.rs [hunk 3]
```diff
@@ -108,22 +111,22 @@ impl StorageMonitorService {
...
 				break
 			};
 		}
 	}
 
 	/// Returns free space in MB, or error if statvfs failed.
-	fn free_space(path: &Path) -> Result<u64, Error> {
-		Ok(fs4::available_space(path).map(|s| s / 1_000_000)?)
+	fn free_space(path: &Path) -> Result<u64> {
+		Ok(fs4::available_space(path).map(|s| s / 1024 / 1024)?)
```
Comment (michalkucharczyk) [inline client/storage-monitor/src/lib.rs:123]:
```suggestion
		Ok(fs4::available_space(path).map(|s| s / 1_000_000)?)
```

File: client/storage-monitor/src/lib.rs (hunk 2 above)
Comment (yjhmelody) [reply client/storage-monitor/src/lib.rs:123]: Well, I think we should use `MiB`(1024 * 1024 Byte). 
This is more accurate for node consumption of storage resources.

File: client/storage-monitor/src/lib.rs (hunk 1 above)
Comment (yjhmelody) [reply client/storage-monitor/src/lib.rs:52]: Personally, I think storage consumption keeps a rhythm with block generation time, so 6 or 12 is better, after all, there is currently no substrate chain with 5s block generation

**Comment by yjhmelody** [comment]
I changed all `MB` to `MiB`.

File: client/storage-monitor/src/lib.rs (hunk 2 above)
Comment (michalkucharczyk) [reply client/storage-monitor/src/lib.rs:123]: ok, for me. but please update docs/comments wherever needed.

File: client/storage-monitor/src/lib.rs [hunk 4]
```diff
@@ -45,19 +48,19 @@ pub struct StorageMonitorParams {
 	#[arg(long = "db-storage-threshold", value_name = "MB", default_value_t = 1000)]
```
Comment (michalkucharczyk) [inline client/storage-monitor/src/lib.rs:48]:
```suggestion
	#[arg(long = "db-storage-threshold", value_name = "MiB", default_value_t = 1000)]
```

**Review by michalkucharczyk — CHANGES_REQUESTED** [review]
Please update docs/comments: `MB` -> `MiB`

**Comment by yjhmelody** [comment]
> Please update docs/comments: `MB` -> `MiB`

Where? I do not know the docs.

**Review by davxy — APPROVED** [review]

File: client/storage-monitor/src/lib.rs [hunk 5]
```diff
@@ -42,22 +45,22 @@ pub struct StorageMonitorParams {
 	/// Required available space on database storage. If available space for DB storage drops below
 	/// the given threshold, node will be gracefully terminated. If `0` is given monitoring will be
 	/// disabled.
-	#[arg(long = "db-storage-threshold", value_name = "MB", default_value_t = 1000)]
+	#[arg(long = "db-storage-threshold", value_name = "MiB", default_value_t = 1000)]
 	pub threshold: u64,
 
-	/// How often available space is polled.
-	#[arg(long = "db-storage-polling-period", value_name = "SECONDS", default_value_t = 5, value_parser = clap::value_parser!(u32).range(1..))]
+	/// How often available space is polled in seconds.
```
Comment (michalkucharczyk) [inline client/storage-monitor/src/lib.rs:51]:
```suggestion
	/// How often available space is polled.
```

Maybe we don't need this?
`--help` is already verbose:
```
      --db-storage-polling-period <SECONDS>
          How often available space is polled
```

**Review by michalkucharczyk — APPROVED** [review]

File: client/storage-monitor/src/lib.rs [hunk 6]
```diff
@@ -42,22 +45,22 @@ pub struct StorageMonitorParams {
 	/// Required available space on database storage. If available space for DB storage drops below
 	/// the given threshold, node will be gracefully terminated. If `0` is given monitoring will be
 	/// disabled.
-	#[arg(long = "db-storage-threshold", value_name = "MB", default_value_t = 1000)]
+	#[arg(long = "db-storage-threshold", value_name = "MiB", default_value_t = 1000)]
 	pub threshold: u64,
 
 	/// How often available space is polled.
-	#[arg(long = "db-storage-polling-period", value_name = "SECONDS", default_value_t = 5, value_parser = clap::value_parser!(u32).range(1..))]
+	#[arg(long = "db-storage-polling-period", value_name = "SECONDS", default_value_t = 6, value_parser = clap::value_parser!(u32).range(1..))]
```
Comment (ggwpez) [inline client/storage-monitor/src/lib.rs:52]: Why was this bumped?

File: client/storage-monitor/src/lib.rs [hunk 7]
```diff
@@ -42,22 +45,22 @@ pub struct StorageMonitorParams {
 	/// Required available space on database storage. If available space for DB storage drops below
 	/// the given threshold, node will be gracefully terminated. If `0` is given monitoring will be
 	/// disabled.
-	#[arg(long = "db-storage-threshold", value_name = "MB", default_value_t = 1000)]
+	#[arg(long = "db-storage-threshold", value_name = "MiB", default_value_t = 1000)]
```
Comment (ggwpez) [inline client/storage-monitor/src/lib.rs:48]: When you start using base 2, may also use `1024` here so the default is one `GiB` and not some odd mixture.

**Review by ggwpez — APPROVED** [review]

File: client/storage-monitor/src/lib.rs (hunk 7 above)
Comment (bkchr) [inline client/storage-monitor/src/lib.rs:48]:
```suggestion
	#[arg(long = "db-storage-threshold", value_name = "MiB", default_value_t = 1024)]
```

File: client/storage-monitor/src/lib.rs (hunk 6 above)
Comment (bkchr) [inline client/storage-monitor/src/lib.rs:52]:
```suggestion
	#[arg(long = "db-storage-polling-period", value_name = "SECONDS", default_value_t = 5, value_parser = clap::value_parser!(u32).range(1..))]
```

**Review by bkchr — APPROVED** [review]

**Comment by bkchr** [comment]
bot merge

**Pull request merged**